more/problem-sets-competitive-programming.md

```diff
@@ -64,6 +65,8 @@
 * [Microcorruption](https://microcorruption.com/login)
 * [Overthewire Wargames fungame to practice CTF](https://overthewire.org/wargames/bandit)
 * [Picoctf](https://picoctf.org/resources.html) (email address *requested*)
+* [ringzer0ctf](https://ringzer0ctf.com/home) (email address *required*)
+* [ROP Wargame Repository](https://github.com/xelenonz/game)
```

Comment:
Can you explain what this site is, and how it's a CTF site?

Reply:
A CTF (Capture the flag) is a security based competition consisting of challenges in various subdomains of security. The ringzer0CTF and echoCTF are CTF sites that keep on growing their challenge repository for people to practice for CTFs.
The ROP Wargame Repository is similiarly a github repository, consisting of challenges based on ROP(Return Oriented Programming) which is used in binary exploitation category of CTFs.

Reply:
You are adding the links in Competitive Site Section.Maybe the link will find its place in another section kindly search through it.

Reply:
> You are adding the links in Competitive Site Section.Maybe the link will find its place in another section kindly search through it.

I added it in capture the flag subcategory. The name of the file indicates competitive programming but the file is no longer only that. As @eshellman pointed in PR [#10556](https://github.com/EbookFoundation/free-programming-books/pull/10556), changing the name of the file might cause conflicts.